packages/story-utils/src/getStories.ts

```diff
@@ -20,7 +28,8 @@ export default function getStories(globPattern: string): StoryData[] {
   return filePaths.flatMap((filePath) => {
     // eslint-disable-next-line @typescript-eslint/no-var-requires
     const storiesFileExports = require(path.resolve(filePath));
-    const { default: metadata, ...storiesMap } = storiesFileExports;
+    const { default: rawMetadata, ...storiesMap } = storiesFileExports;
+    const metadata = metadataSchema.parse(rawMetadata);
```

Comment:
Tested this out in the Along app, and couldn't measure a performance difference.